src/web_service.iced

```diff
@@ -328,6 +328,38 @@ class HackerNewsBinding extends SocialNetworkBinding
 
 #==========================================================================
 
+class GenericSocialBinding extends SocialNetworkBinding
+  constructor : (args) ->
+    @remote_service = args.remote_service
+    if (r = args.name_regexp)? then try @name_regexp = new RegExp(r)
+    super args
+
+  @single_occupancy : () -> false
+  single_occupancy  : () -> GenericSocialBinding.single_occupancy()
+
+  service_name : -> @remote_service
+  proof_type   : -> constants.proof_type.generic_social
+
+  @check_name : (n) ->
+    if not n? or not (n = n.toLowerCase())? then false
+    else if n.match @name_regexp then true
+    else false
+  @name_hint : () -> "alphanumerics, lower case"
```

Comment:
not sure where this is used but since the regex is configurable may as well make it vague but not wrong for now like "a valid username"